### DISCUSSION TITLE
Manifest folder behaviour vs documentation

### DISCUSSION THREAD

**Discussion by MrBlaise** [rancher/rke2#6839] [Q&A]
The documentation [here](https://docs.rke2.io/helm#automatically-deploying-manifests-and-helm-charts) mentions the following:

> Any Kubernetes manifests found in `/var/lib/rancher/rke2/server/manifests` will automatically be deployed to RKE2 in a manner similar to `kubectl apply`, both on startup and when the file is changed on disk. **Deleting files out of this directory will not delete the corresponding resources from the cluster.**

However, I noticed a discrepancy in the behavior when editing files within this directory.

### Steps to Reproduce:
1. Create the following file:
    ```yaml
    # /var/lib/rancher/rke2/server/manifests/testing-namespace.yaml
    apiVersion: v1
    kind: Namespace
    metadata:
      name: testing
    ```
2. The namespace gets created as expected:
    ```shell
    testing               Active   28s
    ```
3. If I remove the file (`rm /var/lib/rancher/rke2/server/manifests/testing-namespace.yaml`), the namespace remains in the cluster, as per the documentation.
4. However, if I **edit the file** and comment out the namespace content, the namespace enters a "Terminating" state:
    ```shell
    testing               Terminating   4m31s
    ```

### Question:
Is this the intended behavior when editing the file? It seems to conflict with the documented expectation that deleting files will not remove the corresponding resources from the cluster. If this is the expected outcome, it might be worth noting this in the documentation to avoid confusion.

**Comment by brandond** [comment]
Yes, this is how it works. The AddOn manifest owns the resources it creates. If you remove resources from the manifest but leave the file in place, the removed resources will be removed from the cluster. If you delete the file from disk, the AddOn remains present in the cluster with its last seen content left in place.

If you want to delete the AddOn after removing it from disk, you can use `kubectl delete addon` to remove it by hand, or --disable it via the server CLI.
